app/lib/search/updater.dart

```diff
@@ -4,92 +4,100 @@
 
 import 'dart:async';
 
-import 'package:gcloud/db.dart';
 import 'package:logging/logging.dart';
 
-import '../frontend/models.dart';
-import '../shared/mock_scores.dart';
 import '../shared/search_service.dart';
-import '../shared/utils.dart' show sliceList;
+import '../shared/task_scheduler.dart';
 
-Logger _logger = new Logger('pub.search.updater');
+import 'backend.dart';
+import 'index_ducene.dart';
 
-class PackageIndexUpdater {
-  final PackageIndex index;
-  Timer _timer;
-  bool _updating = false;
+Logger _logger = new Logger('pub.search.updater');
 
-  PackageIndexUpdater(this.index);
+class IndexUpdateTaskSource implements TaskSource {
+  final BatchIndexUpdater _batchIndexUpdater;
+  DateTime _lastTs;
+  IndexUpdateTaskSource(this._batchIndexUpdater);
 
-  Future update({int limit}) async {
-    if (_updating) {
-      _logger.warning('Update called before previous round completed.');
-      return;
-    }
-    try {
-      _updating = true;
-      await _doUpdate(limit);
-    } catch (e, st) {
-      _logger.severe('Error while updating search index.', e, st);
-    } finally {
-      _updating = false;
+  @override
+  Stream<Task> startStreaming() async* {
+    for (;;) {
+      final DateTime now = new DateTime.now().toUtc();
+      int count = 0;
+      await for (String package
+          in searchBackend.listPackages(updatedAfter: _lastTs)) {
+        count++;
+        yield new Task(package, null);
+      }
+      _batchIndexUpdater.reportScanCount(count);
```

Comment:
Still I find this quite a hack in many ways. It could be removed and just let the indexer re-build the index whenever either one is true:
   * got X new tasks
   * got 1 new task and Y minutes passed since then

But I'll merge it now anyways.